### ISSUE TITLE
recipe for target 'bowtie_local' failed

### ISSUE THREAD

**Issue by jeffpkamp** [nservant/HiC-Pro#284]
When I try to run HiC-Pro I get the following error:

> Bowtie2 alignment step2 ...
> Logs: logs/test1/mapping_step2.log
> Exit: Error in reads alignment - Exit
> /usr/local/bin/HiC-Pro_2.11.1/bin/../scripts//Makefile:98: recipe for target 'bowtie_local' failed
> make: *** [bowtie_local] Error 1

**Comment by jeffpkamp** [comment]
okay... i worked my way past this problem.  now I am getting the error from the pairing step

> /usr/local/anaconda/bin/python /usr/local/bin/HiC-Pro_2.11.1/scripts/mergeSAM.py -q 10 -t -v -f bowtie_results/bwt2/test1/test_R1_hg19.bwt2merged.bam -r bowtie_results/bwt2/test1/test_R2_hg19.bwt2merged.bam -o bowtie_results/bwt2/test1/test_hg19.bwt2pairs.bam
> Traceback (most recent call last):
>   File "/usr/local/bin/HiC-Pro_2.11.1/scripts/mergeSAM.py", line 21, in <module>
>     import pysam
>   File "/uufs/chpc.utah.edu/common/home/u6004424/.local/lib/python2.7/site-packages/pysam/__init__.py", line 5, in <module>
>     from pysam.libchtslib import *
> ImportError: libcrypto.so.10: cannot open shared object file: No such file or directory

It appears to be an issue with the python included with the default package.

**Comment by nservant** [comment]
Hi,
Indeed, I think that you have an installation issue of pysam python package.
Could you try to re-install it ?
Thanks

**Comment by jeffpkamp** [comment]
I found the issue to be a bad interaction between some locally installed python modules and the anaconda ones.  

I now get to this:
> Run quality checks for all samples ...
> Logs: logs/test1/make_Rplots.log
> /usr/local/bin/HiC-Pro_2.11.1/bin/../scripts//Makefile:178: recipe for target 'hic_qc' failed
> make: *** [hic_qc] Error 1

The log doesn't show anything useful, but when I run the script manually it produces the exit code 127 and silently exits without an error.   

one issue I see is bash does not know the exec_cmd program.

I tried circumventing this by just running the R code

I have to run it in a sub shell otherwise it executes an exit command in my shell and boots me out of the singularity container.

```
$: echo $cmd
/usr/local/anaconda/bin/R CMD BATCH --no-save --no-restore "--args picDir='hic_results/pic/test1' hicDir='hic_results/data/test1' statsDir='hic_results/stats/test1' sampleName='test1'" /plot_hic_contacts.R logs/test1/plot_hic_contacts.Rout

$: $(exec $cmd)
/usr/local/anaconda/lib/R/bin/BATCH: 60: /usr/local/anaconda/lib/R/bin/BATCH: cannot create picDir='hic_results/pic/test1': Directory nonexistent

$: ls hic_results/pic/
test1
```

Also I spotted a bug in this script.  Line 47 (and serveral other spots in this sciprt)  should have double square brackets, but currently have single square brackets. For line 47 this  makes it always evaluate to True.

**Comment by nservant** [comment]
Anything in the Rout file ?

**Comment by jeffpkamp** [comment]
Thank you for pointing me to that.  Same issue as the python.  I had to unset all the environmental variables associated with libraries for both python and R and it appears to have finally worked.  

One last thing i ran into that caused a problem.   The `iced` module for python was not being found, so I had to move it to my ~/.local/libs/python2.7/ folder.  Where is this module suppose to be normally?
